src/ray/gcs/redis_module/ray_redis_module.cc

```diff
@@ -505,6 +505,91 @@ int TableLookup_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int
   return REDISMODULE_OK;
 }
 
+// The deleting helper function.
+static Status DeleteKeyHelper(RedisModuleCtx *ctx, RedisModuleString *prefix_str,
+                              RedisModuleString *id_data) {
+  RedisModuleKey *delete_key = nullptr;
+  RAY_RETURN_NOT_OK(
+      OpenPrefixedKey(&delete_key, ctx, prefix_str, id_data, REDISMODULE_READ));
+  if (delete_key == nullptr) {
+    return Status::RedisError("Non-existing key");
+  }
+  auto key_type = RedisModule_KeyType(delete_key);
+  RedisModule_CloseKey(delete_key);
+  if (key_type == REDISMODULE_KEYTYPE_STRING || key_type == REDISMODULE_KEYTYPE_LIST) {
+    // Current Table or Log only has this two types of entries.
+    RAY_RETURN_NOT_OK(
+        OpenPrefixedKey(&delete_key, ctx, prefix_str, id_data, REDISMODULE_WRITE));
+    RedisModule_DeleteKey(delete_key);
+  } else {
+    std::ostringstream ostream;
+    size_t redis_string_size;
+    const char *redis_string_str = RedisModule_StringPtrLen(id_data, &redis_string_size);
+    auto id_binary = std::string(redis_string_str, redis_string_size);
+    ostream << "Undesired type for RAY.TableDelete: " << key_type
+            << " id:" << ray::UniqueID::from_binary(id_binary);
+    RAY_LOG(ERROR) << ostream.str().c_str();
+    return Status::RedisError(ostream.str());
+  }
+  return Status::OK();
+}
+
+/// Delete the redis table associated with a key.
+///
+/// This is called from a client with the command:
+//
+///    RAY.TABLE_DELETE <table_prefix> <pubsub_channel> <id>
+///
+/// \param table_prefix The prefix string for keys in this table.
+/// \param pubsub_channel Unused but follow the interface.
+/// \param id The ID of the key to delete.
+/// \return OK if the deletion succeeds, or an error message string if the
+///         deletion fails.
+int TableDelete_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  if (argc != 4) {
+    return RedisModule_WrongArity(ctx);
+  }
+  RedisModuleString *prefix_str = argv[1];
+  RedisModuleString *id = argv[3];
+  Status status = DeleteKeyHelper(ctx, prefix_str, id);
+  if (status.ok()) {
+    return RedisModule_ReplyWithSimpleString(ctx, "OK");
+  } else {
+    return RedisModule_ReplyWithError(ctx, status.message().c_str());
```

Comment:
instaed of this `if/else` statement, I think we can use the macro `REPLY_AND_RETURN_IF_NOT_OK`